### PR TITLE
Refactor tests

### PR DESCRIPTION
**Pull request by jhkennedy**
This PR refactors the Landsat tests by:
* moving the functions to create mock data to `conftest.py` as fixtures, including:
   * STAC Item creation via `stac_item_factory`
   * HyP3 Job creation via `hyp3_job_factory`
   * HyP3 Batch creation via `hyp3_batch_factory`
   
   This allows more explicit creation of items within any particular test with mock data only containing the parts actually utilized by its-live-monitoring
 * removes `tests/data` as it's all explicitly within the tests now
 * uses a context manager via `unittest.mock.patch` so mocks are cleaned up when no longer needed

Also:
* Adds a pytest GitHub Actions workflow so we run our tests as part of PRs

And, as a bonus, also:
* adds a create-jira-issue GitHub Actions workflow

### CODE REVIEW

**Comment by jhkennedy** [comment]
Note: the diff for `tests/landsat/test_main.py` is rather large and hard to parse. It may be easier to view the file directly to review.

**Review by cirrusasf — APPROVED** [review]
It passes the test. The code is simple and efficient. I approve the PR.
